awkward/util.py

```diff
@@ -199,3 +200,11 @@ class NDArrayOperatorsMixin(object):
             __pos__ = _unary_method(um.positive, 'pos')
             __abs__ = _unary_method(um.absolute, 'abs')
             __invert__ = _unary_method(um.invert, 'invert')
+
+
+# numpy on windows has some strange behavior with dtypes of certain functions
+# requiring us to cast down to int32 for (at least): ufunc.reduceat, repeat
+def windows_safe(array):
+    if os.name == "nt":
+        return array.astype(numpy.int32)
+    return array
```

Comment:
I like this `windows_safe`; I'm going to extend its use to `ufunc.reduceat`, but that could be after this PR.

Reply:
Already did that!

Reply:
Broadening the use of `windows_safe` is something I can do after this PR.

Reply:
Oh, I missed that somehow when I scanned the differences. Thanks!